### DISCUSSION TITLE
SVD dimensions

### DISCUSSION THREAD

**Discussion by EngineerKhan** [jax-ml/jax#8418] [Q&A] [answered]
Asking it out of curiosity. If I apply SVD as:

```
A = jnp.array([[1,1,1],[2,2,2],[3,3,3],[4,4,4]])

a, b, c = linalg.svd(A)
```
Why does it give `b` dimensions as `(3,)` instead of `(3,4)`?

**Comment by jakevdp** [comment, marked as answer]
`b` is returned as a one-dimensional array of singular values, following the standard convention used by NumPy and other numerical computing packages.

I suspect the reason for this is primarily memory savings: if you're doing a large SVD, it's wasteful to return a large matrix that is by definition mostly zeros, when (1) many applications need just the singular values, and (2) it's straightforward to construct the full matrix if needed.
